### ISSUE TITLE
Scheduler does not wake sleeping threads when multiple cores are available

### ISSUE THREAD

**Issue by tlambertz** [hermitcore/libhermit-rs#105]
When using multiple CPU cores, the scheduler does not wake sleeping threads once their sleep duration is over. I guess this happens because of the core-local scheduling, where the main core does not check for threads to be woken on other cores. But I have not verified this.

Simple reproducer:
```
#[cfg(target_os = "hermit")]
extern crate hermit_sys;

use std::thread;
use std::time;

fn main() {
    let mut children = vec![];

    let threadnum = 5;
    for i in 0..threadnum {
        println!("SPAWNING THREAD {}", i);
        children.push(thread::spawn(move || {
            println!("this is thread number {}", i);
            thread::sleep(time::Duration::from_millis(2000));
            println!("---------------THREAD DONE!---------- {}", i);
        }));
    }
    println!("SPAWNED THREADS");

    for child in children {
        let _ = child.join();
    }
}
```

My test project includes a cargo project, makefile and the rusty-loader: [rusty-test-sleep.tar.gz](https://github.com/hermitcore/libhermit-rs/files/5248158/rusty-test-sleep.tar.gz).
The binary is build and started with `make run`. This also sets the correct QEMU options to reproduce the bug.

**Comment by stlankes** [comment]
yeah, I think that this is the reason for issue #78

**Comment by stlankes** [comment]
@tlambertz I think that I solved with #106 the issue. Unfortunately, #78 has another reason.

**Comment by tlambertz** [comment]
Yep, I can confirm the issue is fixed.
